frontend/js/modules/search.js

```diff
@@ -0,0 +1,179 @@
+module.exports = (function () {
+    'use strict';
+    // Global dependencies
+    var m  = require('mithril');
+    var ld = require('lodash');
+  
+    // Local dependencies
+    var conf      = require('../configuration.js');
+    var auth      = require('../auth.js');
+    var u         = auth.userInfo;
+    var layout    = require('./layout.js');
+    var model     = require('../model/group.js');
+  
+    var search = {};
+  
+    /**
+    * ## Controller
+    *
+    * Used for module state and actions.
+    */
+  
+    search.controller = function () {
+      if (!auth.isAuthenticated()) {
+        conf.unauthUrl(true);
+        return m.route('/login');
+      }
+      document.title = conf.LANG.SEARCH.TITLE + ' - ' + conf.SERVER.title;
+  
+      var c = {};
+  
+      /**
+      * ### computeSearchResults
+      *
+      * `computeSearchResults` is an internal function that gathers search groups
+      * and pads.
+      */
+  
+      c.computeSearchResults = function () {
+
+        var items  = function (data) {
+          return  ld(data)
+            .values()
+            //.sortBy('name')
+            .value();
+        };
+        if(c.results != null){
+          c.searchResults = {
+            pads: items(c.results.pads)
+          };
+        }
+        else{
+          c.searchResults = {
+            pads: []
+          };
+        }
+      };
+  
+
+
+    c.search       = m.prop('');
+    c.filterSearch = function () {
+      m.request({
+        method: 'GET',
+        url: conf.URLS.PAD + '/search?q=' + encodeURI(c.search()),
```

Comment:
Instead of manually generating the queryString, try looking into using options.params instead.  See: https://mithril.js.org/request.html